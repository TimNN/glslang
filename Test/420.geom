#version 150 core

// testing input arrays without a gl_in[] block redeclaration, see 400.geom for with

int i;

void foo()
{
    gl_in.length();  // ERROR
    gl_in[1].gl_Position;
    gl_in[i].gl_Position;  // ERROR
}

layout(triangles) in;

in vec4 color3[3];

void foo3()
{
    gl_in.length();
    gl_in[i].gl_Position;
    color3.length();
}
