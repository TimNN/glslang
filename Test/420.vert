#version 420 core

varying vec2 v2;               // ERROR, varying reserved
in vec4 bad[10];
highp in vec4 badorder;
out invariant vec4 badorder2;
in centroid vec4 badorder4;    // ERROR, no centroid input to vertex stage
out flat vec4 badorder3;
void bar(in const float a);
void bar2(highp in float b);
smooth flat out vec4 rep;      // ERROR, replicating interpolation qualification
centroid sample out vec4 rep2; // ERROR, replicating auxiliary qualification
in uniform vec4 rep3;          // ERROR, replicating storage qualification

void main()
{
    int i;
    if (i == 3)
        int j = i;
    else
        int k = j;              // ERROR, j is undeclared
    int m = k;                  // ERROR, k is undeclared
    int n = j;                  // ERROR, j is undeclared

    while (true)
        int jj;
    int kk = jj;                // ERROR, jj is undeclared
}

void bar(in highp volatile vec4 v)
{
}