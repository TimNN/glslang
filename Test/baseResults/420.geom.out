Warning, version 150 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:9: 'length' :  array must be declared with a size before using this method
ERROR: 0:11: '[' :  array must be redeclared with a size before being indexed with a variable
ERROR: 2 compilation errors.  No code generated.

invocations = 0
max_vertices = 0
input primitive = triangles
output primitive = none
ERROR: node is still EOpNull!
0:7  Function Definition: foo( (void)
0:7    Function Parameters: 
0:9    Sequence
0:9      Constant:
0:9        1 (const int)
0:10      gl_Position: direct index for structure (4-component vector of float)
0:10        direct index (in block{gl_Position,gl_PointSize,gl_ClipDistance})
0:10          'gl_in' (in 3-element array of block{gl_Position,gl_PointSize,gl_ClipDistance})
0:10          Constant:
0:10            1 (const int)
0:10        Constant:
0:10          0 (const int)
0:11      gl_Position: direct index for structure (4-component vector of float)
0:11        indirect index (in block{gl_Position,gl_PointSize,gl_ClipDistance})
0:11          'gl_in' (in 3-element array of block{gl_Position,gl_PointSize,gl_ClipDistance})
0:11          'i' (int)
0:11        Constant:
0:11          0 (const int)
0:18  Function Definition: foo3( (void)
0:18    Function Parameters: 
0:20    Sequence
0:20      Constant:
0:20        3 (const int)
0:21      gl_Position: direct index for structure (4-component vector of float)
0:21        indirect index (in block{gl_Position,gl_PointSize,gl_ClipDistance})
0:21          'gl_in' (in 3-element array of block{gl_Position,gl_PointSize,gl_ClipDistance})
0:21          'i' (int)
0:21        Constant:
0:21          0 (const int)
0:22      Constant:
0:22        3 (const int)
0:?   Linker Objects
0:?     'i' (int)
0:?     'gl_in' (in 3-element array of block{gl_Position,gl_PointSize,gl_ClipDistance})
0:?     'color3' (in 3-element array of 4-component vector of float)

